analyses/tmb-compare-tcga/compare-tmb.Rmd

```diff
@@ -0,0 +1,273 @@
+---
+title: "Tumor Mutation Burden Comparison with TCGA data"
+output: 
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+author: C. Savonen for ALSF CCDL
+date: 2019
+---
+
+#### Purpose
+
+Plot the consensus tumor mutation burden statistics for PBTA in comparison to TCGA brain-related data. 
+
+#### Usage
+
+To run this from the command line, use:
+```
+Rscript -e "rmarkdown::render('analyses/tmb-compare-tcga/compare-tmb.Rmd', 
+                              clean = TRUE)"
+```
+
+_This assumes you are in the top directory of the repository._
+
+## Setup
+
+#### Packages and functions
+
+```{r}
+if (!("TCGAbiolinks" %in% installed.packages())) {
+  install.packages("TCGAbiolinks")
+}
+
+# magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Load in these functions so we can use `maf_to_granges`
+source(file.path("..", "snv-callers", "util", "wrangle_functions.R"))
+```
+
+Declare names of input and output directories.
+
+```{r}
+data_dir <- file.path("..", "..", "data")
+scratch_dir <- file.path("..", "..", "scratch")
+results_dir <- "results"
+plots_dir <- "plots"
+```
+
+Create output directories for this analysis. 
+
+```{r}
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+```
+
+Custom function for plotting the TMB. 
+
+```{r}
+sina_plot <- function(data, colour = NULL, title) {
+  # Given a data.frame with `short_histology` and `tmb` information, make a sina plot from it
+  #
+  # Args:
+  #
+  # data : a data.frame with `short_histology` and `tmb` as columns
+  # colour: the color you would like the points to be
+  # title: The title for the plot. To be passed to `ggtitle`
+  #
+  # Returns:
+  #   A sina plot by histology group
+  data %>%
+    ggplot2::ggplot(ggplot2::aes(x = reorder(short_histology, tmb, mean), y = log10(tmb))) +
+    ggforce::geom_sina(color = colour, alpha = 0.3) +
```

Comment:
Is there a parameter to change the width here? Some of the histologies with less samples look a little strange in the PNG

Reply:
Not a big deal if not -- only one histology really caught my eye.